typewise_alert.py

```diff
@@ -1,46 +1,45 @@
 
+CoolingTypeLimit = { 'PASSIVE_COOLING': {'lowerLimit': 0,'upperLimit':35},
+                     'HI_ACTIVE_COOLING': {'lowerLimit': 0,'upperLimit':45},
+                     'MED_ACTIVE_COOLING': {'lowerLimit': 0,'upperLimit':40}
+                   }
+
+EMailInfo = { 'TOO_LOW' : {'Recepient':'a.b@c.com','Message':'Hi, the temperature is too low'},
+              'TOO_HIGH' : {'Recepient':'a.b@c.com','Message':'Hi, the temperature is too high'}
+            }
+
 def infer_breach(value, lowerLimit, upperLimit):
   if value < lowerLimit:
     return 'TOO_LOW'
   if value > upperLimit:
     return 'TOO_HIGH'
   return 'NORMAL'
 
-
 def classify_temperature_breach(coolingType, temperatureInC):
-  lowerLimit = 0
-  upperLimit = 0
-  if coolingType == 'PASSIVE_COOLING':
-    lowerLimit = 0
-    upperLimit = 35
-  elif coolingType == 'HI_ACTIVE_COOLING':
-    lowerLimit = 0
-    upperLimit = 45
-  elif coolingType == 'MED_ACTIVE_COOLING':
-    lowerLimit = 0
-    upperLimit = 40
-  return infer_breach(temperatureInC, lowerLimit, upperLimit)
+  if coolingType in CoolingTypeLimit.keys():
+     coolingType_range = CoolingTypeLimit[coolingType]
+     return infer_breach(temperatureInC, coolingType_range['lowerLimit'], coolingType_range['upperLimit'])
+  else:
+    return False
 
 
 def check_and_alert(alertTarget, batteryChar, temperatureInC):
-  breachType =\
-    classify_temperature_breach(batteryChar['coolingType'], temperatureInC)
-  if alertTarget == 'TO_CONTROLLER':
-    send_to_controller(breachType)
-  elif alertTarget == 'TO_EMAIL':
-    send_to_email(breachType)
-
+  breachType = classify_temperature_breach(batteryChar['coolingType'], temperatureInC)
+  if alertTarget in Alert_Target.keys():
+    Alert_Target[alertTarget](breachType)
+    return True
+  return False
 
 def send_to_controller(breachType):
-  header = 0xfeed
-  print(f'{header}, {breachType}')
-
+  print(f'BreachType: {breachType}')
 
 def send_to_email(breachType):
-  recepient = "a.b@c.com"
-  if breachType == 'TOO_LOW':
-    print(f'To: {recepient}')
-    print('Hi, the temperature is too low')
-  elif breachType == 'TOO_HIGH':
-    print(f'To: {recepient}')
-    print('Hi, the temperature is too high')
+  print(f'{EMailInfo[breachType]}')
+
+def send_to_console(breachType):
+  print(f'BreachType: {breachType}')
+  
+Alert_Target = { 'TO_CONTROLLER': send_to_controller,
```

Comment:
If you make a mistake/bug here (sending to email on the TO_CONTROLLER line instead of sending to the controller), your tests will still pass. Same if the 'to' field was left blank in the email...
So even with passing tests, some parts of your code need to be proven in an integrated environment.
How can you prove all of your code before integrating it with 3rd party packages like emailers, controllers, etc.?